pkg/providers/vsphere/vmprovider_vm_test.go

```diff
@@ -2825,6 +2826,262 @@ func vmTests() {
+					BeforeEach(func() {
+						vm.Spec.CurrentSnapshot = &vmopv1common.LocalObjectRef{
+							APIVersion: vmSnapshot.APIVersion,
+							Kind:       vmSnapshot.Kind,
+							Name:       "non-existent-snapshot",
+						}
+					})
+
+					It("should not trigger a revert (create snapshot workflow)", func() {
+						// Create a different snapshot CR
+						nonExistentSnapshot := builder.DummyVirtualMachineSnapshot("", "non-existent-snapshot", vm.Name)
+						nonExistentSnapshot.Namespace = nsInfo.Namespace
+						Expect(ctx.Client.Create(ctx, nonExistentSnapshot)).To(Succeed())
+
+						err := createOrUpdateVM(ctx, vmProvider, vm)
+						Expect(err).ToNot(HaveOccurred())
+
+						// Verify VM status reflects current snapshot
+						Expect(vm.Status.CurrentSnapshot).ToNot(BeNil())
+						Expect(vm.Status.CurrentSnapshot.Name).To(Equal(vm.Spec.CurrentSnapshot.Name))
+					})
+				})
+
+				Context("when desired snapshot CR doesn't exist", func() {
+					BeforeEach(func() {
+						vm.Spec.CurrentSnapshot = &vmopv1common.LocalObjectRef{
+							APIVersion: vmSnapshot.APIVersion,
+							Kind:       vmSnapshot.Kind,
+							Name:       vmSnapshot.Name,
+						}
+					})
+
+					It("should fail with snapshot CR not found error", func() {
+						err := createOrUpdateVM(ctx, vmProvider, vm)
+						Expect(err).To(HaveOccurred())
+						Expect(err.Error()).To(ContainSubstring("virtualmachinesnapshots.vmoperator.vmware.com \"test-revert-snap\" not found"))
+					})
+				})
+
+				Context("when VM is already at desired snapshot", func() {
+					It("should succeed without performing revert", func() {
+						// Create VM first to get vcVM reference
+						vcVM, err := createOrUpdateAndGetVcVM(ctx, vmProvider, vm)
+						Expect(err).ToNot(HaveOccurred())
+
+						// Create snapshot in vCenter
+						task, err := vcVM.CreateSnapshot(ctx, vmSnapshot.Name, "test snapshot for revert", false, false)
+						Expect(err).ToNot(HaveOccurred())
+						Expect(task.Wait(ctx)).To(Succeed())
+
+						// Create snapshot CR for the above snapshot
+						Expect(ctx.Client.Create(ctx, vmSnapshot)).To(Succeed())
+
+						// Set desired snapshot to point to the above snapshot
+						vm.Spec.CurrentSnapshot = &vmopv1common.LocalObjectRef{
+							APIVersion: vmSnapshot.APIVersion,
+							Kind:       vmSnapshot.Kind,
+							Name:       vmSnapshot.Name,
+						}
+
+						err = createOrUpdateVM(ctx, vmProvider, vm)
+						Expect(err).ToNot(HaveOccurred())
+
+						// Verify VM status reflects current snapshot
+						Expect(vm.Status.CurrentSnapshot).ToNot(BeNil())
+						Expect(vm.Status.CurrentSnapshot.Name).To(Equal(vmSnapshot.Name))
+					})
+				})
+
+				Context("when reverting to valid snapshot", func() {
+					var secondSnapshot *vmopv1.VirtualMachineSnapshot
+
+					It("should successfully revert to desired snapshot", func() {
+						// Create VM first
+						var err error
+						vcVM, err = createOrUpdateAndGetVcVM(ctx, vmProvider, vm)
```

Comment:
There is a mix of this and `vcVM, err := ...` in these tests. I don't think we need the outer `vcVM` and can use `:=` here & elsewhere